src/vm/hints/find_element_hint.rs

```diff
@@ -106,115 +34,132 @@ pub fn find_element(
+                    return Err(VirtualMachineError::InvalidIndex(
+                        find_element_index_value.clone(),
+                        maybe_rel_key.clone(),
+                        found_key.clone(),
+                    ));
+                }
 
-                    if !elm_size.is_positive() {
-                        return Err(VirtualMachineError::ValueOutOfRange(elm_size.clone()));
-                    }
+                vm.memory
+                    .insert(
+                        &index_addr,
+                        &MaybeRelocatable::Int(find_element_index_value.clone()),
+                    )
+                    .map_err(VirtualMachineError::MemoryError)?;
+
+                vm.exec_scopes.delete_variable("find_element_index");
+                Ok(())
+            } else {
+                let n_elms = if let MaybeRelocatable::Int(ref n_elms) = maybe_rel_n_elms {
+                    n_elms
+                } else {
+                    return Err(VirtualMachineError::ExpectedInteger(
+                        maybe_rel_n_elms.clone(),
+                    ));
+                };
 
-                    if let Some(find_element_index_value) = vm.find_element_index.clone() {
-                        vm.find_element_index = None;
-                        let array_start = vm
-                            .memory
-                            .get(&array_ptr_addr)
-                            .map_err(VirtualMachineError::MemoryError)?
-                            .ok_or(VirtualMachineError::FindElemNoFoundKey)?;
-
-                        let found_key =
-                            vm.memory
-                                .get(&array_start.add_int_mod(
-                                    &(elm_size * &find_element_index_value),
-                                    &vm.prime,
-                                )?)
-                                .map_err(VirtualMachineError::MemoryError)?
-                                .ok_or(VirtualMachineError::FindElemNoFoundKey)?;
-
-                        if found_key != maybe_rel_key {
-                            return Err(VirtualMachineError::InvalidIndex(
-                                find_element_index_value,
-                                maybe_rel_key.clone(),
-                                found_key.clone(),
+                if n_elms.is_negative() {
+                    return Err(VirtualMachineError::ValueOutOfRange(n_elms.clone()));
+                }
+
+                if let Some(variables) = vm.exec_scopes.get_local_variables() {
+                    if let Some(PyValueType::BigInt(find_element_max_size)) =
+                        variables.get("find_element_max_size")
+                    {
+                        if n_elms > find_element_max_size {
+                            return Err(VirtualMachineError::FindElemMaxSize(
+                                find_element_max_size.clone(),
+                                n_elms.clone(),
```

Comment:
Shouldn't this be tested unconditionally at the beginning of the hint?

Reply:
Apparently not https://github.com/starkware-libs/cairo-lang/blob/2abd303e1808612b724bc1412b2b5babd04bb4e7/src/starkware/cairo/common/find_element.cairo#L48-L51
It's tested only if there is an element_max_size

Reply:
Sorry, I meant unconditionally regarding whether or not an element index was passed.